src/containers/ResourceList/ResourceList.js

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2019 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { Component } from 'react';
+
+import {
+  InlineNotification,
+  StructuredListBody,
+  StructuredListCell,
+  StructuredListHead,
+  StructuredListRow,
+  StructuredListSkeleton,
+  StructuredListWrapper
+} from 'carbon-components-react';
+import { Link } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { getErrorMessage } from '../../utils';
+import { ALL_NAMESPACES } from '../../constants';
+import { getCustomResources } from '../../api';
+import { getSelectedNamespace } from '../../reducers';
+
+class ResourceListContainer extends Component {
+  state = {
+    loading: true,
+    resources: []
+  };
+
+  componentDidMount() {
+    const { group, version, type } = this.props.match.params;
+    const { namespace } = this.props;
+    this.fetchResources(group, version, type, namespace);
+  }
+
+  componentDidUpdate(prevProps) {
+    const { match, namespace } = this.props;
+    const { group, version, type } = match.params;
+    const { match: prevMatch, namespace: prevNamespace } = prevProps;
+    const {
+      type: prevType,
+      group: prevGroup,
+      version: prevVersion
+    } = prevMatch.params;
+
+    if (
+      namespace !== prevNamespace ||
+      type !== prevType ||
+      group !== prevGroup ||
+      version !== prevVersion
+    ) {
+      this.fetchResources(group, version, type, namespace);
+    }
+  }
+
+  fetchResources(group, version, type, namespace) {
+    return getCustomResources(group, version, type, { namespace }).then(
+      resources => {
+        this.setState({
+          loading: false,
+          resources
+        });
+      }
+    );
+  }
+
+  render() {
+    const { match, namespace: selectedNamespace } = this.props;
+    const { type } = match.params;
+    const { error, loading, resources } = this.state;
+
+    if (loading) {
+      return <StructuredListSkeleton border />;
+    }
+
+    if (error) {
+      return (
+        <InlineNotification
+          kind="error"
+          hideCloseButton
+          lowContrast
+          title={`Error loading ${type}`}
+          subtitle={getErrorMessage(error)}
+        />
+      );
+    }
+
+    return (
+      <>
```

Comment:
no need for the fragment here since there's just a single root component being returned